MAINT Update requirements and add some documentation addtions

jjerphan opened the pull request:
A few little improvements:
 - jupyter notebook are mentioned but not part of dependency. This adds jupyter-lab for user friendlyness.
 - `sns.histplot` is not working on latest versions of matplotlib which are excluded by this PR. See: https://github.com/mwaskom/seaborn/issues/3072
 - the `README` has been updated accordingly
 - add an expected output when `ramp-test`ing submission

jjerphan committed:
Add jupyter-lab as a dependency and exclude matplotlib>=3.6.1

jupyter notebook are mentioned but not part of dependency.

This commit add jupyter-lab for user friendlyness.

Also `sns.histplot` is not working on latest versions
of matplotlib which are excluded by this commit.

See: https://github.com/mwaskom/seaborn/issues/3072

jjerphan committed:
DOC Document expected output when testing the submission example

rth commented on requirements.txt:
```diff
@@ -1,7 +1,9 @@
 scikit-learn
 folium
 ramp-workflow
+jupyter-lab
```
It shouldn't be part of the requirements, because this is also used to create the env on the server. They should install jupyterlab separately (maybe it could be indicated in the readme).

rth reviewed (commented):
Thanks could you make this PR to the upstream repo instead (and cc me)? Thanks!

Minor comment otherwise LGTM.

jjerphan committed:
Remove jupyterlab from requirements and instead document it

jjerphan commented:
Thanks!

I just have opened https://github.com/ramp-kits/bike_counters/pull/4 and addressed your comment in https://github.com/rth/bike_counters/pull/1#discussion_r996812458.
